trick_source/sim_services/JSONVariableServer/JSONVariableServer.cpp

```diff
@@ -25,7 +25,7 @@ Trick::JSONVariableServer::JSONVariableServer() :
 
 Trick::JSONVariableServer::~JSONVariableServer() {
     if ( listen_dev.hostname != NULL ) {
-        free(listen_dev.hostname) ;
+        free((char*)listen_dev.hostname) ;
```

Comment:
Why is this necessary? `listen_dev.hostname` is already a `char*`.

Reply:
The free or the cast to a char*?

Reply:
Good catch. I have no idea why I initially did that. I'll fix it.

Reply:
I remember now! I did that because I changed listen_dev.hostname to a const char* and it was throwing an error when I tried to free a constant.